File: leetcode/AIcoding/1.Gemini-2.5/javascript_solutions/LCR 034-lwyVBB.js
var isAlienSorted = function(words, order) {
    const orderMap = new Array(26);
    for (let i = 0; i < order.length; i++) {
        orderMap[order.charCodeAt(i) - 'a'.charCodeAt(0)] = i;
    }

    for (let i = 0; i < words.length - 1; i++) {
        const word1 = words[i];
        const word2 = words[i + 1];

        let foundDifference = false;
        for (let j = 0; j < Math.min(word1.length, word2.length); j++) {
            const char1Rank = orderMap[word1.charCodeAt(j) - 'a'.charCodeAt(0)];
            const char2Rank = orderMap[word2.charCodeAt(j) - 'a'.charCodeAt(0)];

            if (char1Rank > char2Rank) {
                return false;
            }
            if (char1Rank < char2Rank) {
                foundDifference = true;
                break;
            }
        }

        if (!foundDifference && word1.length > word2.length) {
            return false;
        }
    }

    return true;
};

File: leetcode/AIcoding/4.GPT-4o/c_solutions/LCR 034-lwyVBB.c
#include <stdbool.h>
#include <string.h>

bool isAlienSorted(char **words, int wordsSize, char *order) {
    int orderMap[26];
    for (int i = 0; i < 26; i++) {
        orderMap[order[i] - 'a'] = i;
    }

    for (int i = 1; i < wordsSize; i++) {
        int j = 0;
        while (j < strlen(words[i - 1]) && j < strlen(words[i]) && words[i - 1][j] == words[i][j]) {
            j++;
        }
        if (j < strlen(words[i - 1]) && (j == strlen(words[i]) || orderMap[words[i - 1][j] - 'a'] > orderMap[words[i][j] - 'a'])) {
            return false;
        }
    }
    return true;
}